index.js

```diff
@@ -1,43 +1,20 @@
-// utils
-function $(selector) {
-  return document.getElementById(selector);
-}
-
-const $taskInput = $('#taskInput');
-const $taskList = $('#taskList');
-
-const handleKeydownTaskInput = (event) => {
-  const taskText = $taskInput.value.trim();
-
-  if (event.key !== 'Enter' || !taskText) {
-    return;
-  }
-
-  const taskItem = createTaskItemElement(taskText);
-  $taskList.appendChild(taskItem);
-  $taskInput.value = '';
-};
-
-$taskInput.addEventListener('keydown', handleKeydownTaskInput);
-
-const createTaskItemElement = (text) => {
-  const taskItemLi = document.createElement('li');
-  const taskTextSpan = document.createElement('span');
-  const deleteButtonImg = document.createElement('img');
-  const svgImage =
-    '<svg fill="#af5b5e" width="40px" height="40px" viewBox="-3.5 0 19 19" xmlns="http://www.w3.org/2000/svg" class="cf-icon-svg"><g id="SVGRepo_bgCarrier" stroke-width="0"></g><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g><g id="SVGRepo_iconCarrier"><path d="M11.383 13.644A1.03 1.03 0 0 1 9.928 15.1L6 11.172 2.072 15.1a1.03 1.03 0 1 1-1.455-1.456l3.928-3.928L.617 5.79a1.03 1.03 0 1 1 1.455-1.456L6 8.261l3.928-3.928a1.03 1.03 0 0 1 1.455 1.456L7.455 9.716z"></path></g></svg>';
-
-  taskTextSpan.textContent = text;
-  deleteButtonImg.src = 'data:image/svg+xml,' + encodeURIComponent(svgImage);
-
-  const handleClickDeleteButtonImg = () => {
-    $taskList.removeChild(taskItemLi);
-  };
-
-  deleteButtonImg.addEventListener('click', handleClickDeleteButtonImg);
-
-  taskItemLi.appendChild(taskTextSpan).classList.add('add');
-  taskItemLi.appendChild(deleteButtonImg).classList.add('deleteBtn');
-
-  return taskItemLi;
-};
+import { $ } from './src/utils/dom.js';
+import { todos } from './src/models/todos.js'
+import { addTodo, renderList } from './src/components/todo.js'
+
+// input dom
+const $todoInput = $('#todoInput');
+// 필터 기능 테스트하려고 만든 임의의 버튼
+const $allTodo = $('#allTodo');
+const $leftTodo = $('#leftTodo');
+const $completedTodo = $('#completedTodo');
+const $completedAll = $('#completedAll');
+
+// input에 입력을 하면 todo추가
+$todoInput.addEventListener('keyup', addTodo);
+// todos는 모든 todo를 render
+$allTodo.addEventListener('click', ()=>{renderList(todos)})
+// 필터링
+$leftTodo.addEventListener('click', ()=>{renderList('unCheckedTodos')})
+$completedTodo.addEventListener('click', () =>{renderList('checkedTodos')})
+$completedAll.addEventListener('click', () =>{renderList('checkedAll')})
```

Comment:
p3. 코드 스타일 관련해서는 이 정도를 고려해보면 좋을 것 같아요~

1. 개행을 추가한다.
2. 만약 todo 앱을 이용하던 중에 에러가 발생하면 에러를 처리해주기 위해 try-catch문을 사용한다.
3. todo 앱을 시작하기 위해 초기화 한다는 의미에서 init과 같은 함수명으로 index.js 파일내의 기능들을 묶어준다.

```suggestion
import { addTodo, renderList } from './src/components/todo.js';

import { todos } from './src/models/todos.js';

import { $ } from './src/utils/dom.js';

function init() {
  try {
    const $todoInput = $('#todoInput');
    const $allTodo = $('#allTodo');
    const $leftTodo = $('#leftTodo');
    const $completedTodo = $('#completedTodo');
    const $completedAll = $('#completedAll');

    $todoInput.addEventListener('keyup', addTodo);
    $allTodo.addEventListener('click', () => {
      renderList(todos);
    });
    $leftTodo.addEventListener('click', () => {
      renderList('unCheckedTodos');
    });
    $completedTodo.addEventListener('click', () => {
      renderList('checkedTodos');
    });
    $completedAll.addEventListener('click', () => {
      renderList('checkedAll');
    });
  } catch (error) {
    window.alert(error);
  }
}

init();

```